.github/workflows/security-scan-weekly.yaml

```diff
@@ -0,0 +1,109 @@
+name: Security Scan Weekly
+
+on:
+  schedule:
+    - cron: "0 0 * * 1" # Runs every Monday at 00:00 UTC.
+  workflow_dispatch:
+
+jobs:
+  repo-security-scan:
+    runs-on: [ubuntu-latest]
+    name: Repo Security Scan
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      #########
+      # Trivy #
+      #########
+      - name: Run Trivy vulnerability scanner in repo mode
+        uses: aquasecurity/trivy-action@0.29.0
+        with:
+          scan-type: 'fs'
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+        env:
+          TRIVY_DB_REPOSITORY: ghcr.io/canonical/comsys-build-tools/trivy-db:2
```

Comment:
Probably not the best mirror to be using but we can fix that later I suppose. I went and checked and it hasn't been updated in the last 5 months because "this repo has been inactive for 60 days".
Maybe you can look at an alternative?